### PR TITLE
Add the nomad snap

### PR DESCRIPTION
**Pull request by come-maiz**
This change is [Reviewable](https://reviewable.io/reviews/ubuntu/snappy-playpen/126)

<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ubuntu/snappy-playpen/pull/126%23discussion_r68903891%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/126%23discussion_r68905466%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/126%23discussion_r68942933%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/126%23discussion_r68970935%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%206c399b920da956a39cdacb230d22baa83fda63bf%20nomad/parts/plugins/x-nomad.py%2033%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/126%23discussion_r68903891%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40sergiusens%20%40kyrofa%3A%20what%20about%20this%20one%3F%20%5E%5Cr%5CnThe%20go%20plugin%20is%20not%20easy%20to%20extend.%20So%20what%20is%20the%20recommendation%20here%3F%20Copy%20the%20full%20plugin%20and%20inherit%20from%20BasePlugin%2C%20or%20do%20this%20awful%20inheritance%20trick%20to%20skip%20the%20parent%3F%20We%20could%20also%20do%20multiple%20inheritance%2C%20but%20I%20think%20that%27s%20a%20bigger%20mess.%22%2C%20%22created_at%22%3A%20%222016-06-29T08%3A23%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/617831%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elopio%22%7D%7D%2C%20%7B%22body%22%3A%20%22Note%20as%20well%20https%3A//bugs.launchpad.net/snapcraft/%2Bbug/1595964%20and%20linked%20example%20%28https%3A//github.com/ubuntu/snappy-playpen/blob/master/vlc/parts/plugins/x_vlcautotools.py%29%22%2C%20%22created_at%22%3A%20%222016-06-29T08%3A34%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1823296%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/didrocks%22%7D%7D%2C%20%7B%22body%22%3A%20%22First%20of%20all%2C%20shouldn%27t%20that%20be%20%60super%28go.GoPlugin%2C%20self%29.build%28%29%60%20%3Astuck_out_tongue%3A%20%3F%5Cr%5Cn%5Cr%5CnYikes%2C%20so%20this%20thing%20requires%20golang%20and%20a%20go%20workspace%2C%20but%20then%20builds%20with%20make%3F%20First%20of%20all%2C%20I%20don%27t%20really%20have%20a%20problem%20with%20the%20inheritance%20skip%20like%20that%20if%20it%20reduces%20duplication.%20But%20does%20nomad%20need%20something%20really%20special%20from%20the%20go%20plugin%20%28e.g.%20the%20importpath%20support%2C%20or%20go-packages%29%3F%20Or%20really%20just%20the%20setup%20of%20the%20workspace%3F%20If%20the%20latter%2C%20I%20say%20scratch%20the%20inheritance%20and%20write%20your%20own%20pull%20step.%20You%20can%20better%20guarantee%20its%20functionality%2C%20and%20make%20it%20work%20specifically%20for%20nomad.%22%2C%20%22created_at%22%3A%20%222016-06-29T13%3A14%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/946674%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kyrofa%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20it%20needs%20from%20the%20go%20plugin%20the%20init%2C%20env%2C%20pull%20and%20clean_pull.%20I%20can%20copy%20all%20that%2C%20I%27m%20just%20not%20sure%20what%27s%20better.%22%2C%20%22created_at%22%3A%20%222016-06-29T15%3A37%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/617831%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elopio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20nomad/parts/plugins/x-nomad.py%3AL1-50%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 6c399b920da956a39cdacb230d22baa83fda63bf nomad/parts/plugins/x-nomad.py 33'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/126#discussion_r68903891'>File: nomad/parts/plugins/x-nomad.py:L1-50</a></b>
- <a href='https://github.com/elopio'></a> @sergiusens @kyrofa: what about this one? ^
  The go plugin is not easy to extend. So what is the recommendation here? Copy the full plugin and inherit from BasePlugin, or do this awful inheritance trick to skip the parent? We could also do multiple inheritance, but I think that's a bigger mess.
- <a href='https://github.com/didrocks'></a> Note as well https://bugs.launchpad.net/snapcraft/+bug/1595964 and linked example (https://github.com/ubuntu/snappy-playpen/blob/master/vlc/parts/plugins/x_vlcautotools.py)
- <a href='https://github.com/kyrofa'></a> First of all, shouldn't that be `super(go.GoPlugin, self).build()` :stuck_out_tongue: ?
  Yikes, so this thing requires golang and a go workspace, but then builds with make? First of all, I don't really have a problem with the inheritance skip like that if it reduces duplication. But does nomad need something really special from the go plugin (e.g. the importpath support, or go-packages)? Or really just the setup of the workspace? If the latter, I say scratch the inheritance and write your own pull step. You can better guarantee its functionality, and make it work specifically for nomad.
- <a href='https://github.com/elopio'></a> I think it needs from the go plugin the init, env, pull and clean_pull. I can copy all that, I'm just not sure what's better.

<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/126?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/126?approve=1'></a>&nbsp;<a href='https://github.com/ubuntu/snappy-playpen/pull/126'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: nomad/parts/plugins/x-nomad.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import shutil
+
+from snapcraft import BasePlugin
+from snapcraft.plugins import go
+
+
+class NomadPlugin(go.GoPlugin):
+
+    def build(self):
+        BasePlugin.build(self)
```
Comment (elopio) [inline nomad/parts/plugins/x-nomad.py]: @sergiusens @kyrofa: what about this one? ^
The go plugin is not easy to extend. So what is the recommendation here? Copy the full plugin and inherit from BasePlugin, or do this awful inheritance trick to skip the parent? We could also do multiple inheritance, but I think that's a bigger mess.

Comment (didrocks) [inline nomad/parts/plugins/x-nomad.py]: Note as well https://bugs.launchpad.net/snapcraft/+bug/1595964 and linked example (https://github.com/ubuntu/snappy-playpen/blob/master/vlc/parts/plugins/x_vlcautotools.py)

Comment (kyrofa) [inline nomad/parts/plugins/x-nomad.py]: First of all, shouldn't that be `super(go.GoPlugin, self).build()` :stuck_out_tongue: ?

Yikes, so this thing requires golang and a go workspace, but then builds with make? First of all, I don't really have a problem with the inheritance skip like that if it reduces duplication. But does nomad need something really special from the go plugin (e.g. the importpath support, or go-packages)? Or really just the setup of the workspace? If the latter, I say scratch the inheritance and write your own pull step. You can better guarantee its functionality, and make it work specifically for nomad.

Comment (elopio) [inline nomad/parts/plugins/x-nomad.py]: I think it needs from the go plugin the init, env, pull and clean_pull. I can copy all that, I'm just not sure what's better.

**Pull request closed without merging**